src/legacy/ui/public/index_patterns/_field_list.ts

```diff
@@ -18,16 +18,21 @@
  */
 
 import { IndexedArray } from 'ui/indexed_array';
-import { Field } from './_field';
+import { IndexPattern } from 'ui/index_patterns/_index_pattern';
+import { Field, FieldSpec } from './_field';
 
-export class FieldList extends IndexedArray {
-  constructor(indexPattern, specs) {
+export class FieldList extends IndexedArray<Field> {
+  // makes typescript work, but breaks kibana
+  //public byType: Record<string, Field[]> = {};
+  //public byName: Record<string, Field> = {};
```

Comment:
It's because these should really be in the `IndexedArray` interface. I'll update this.